### PR TITLE
DOC: explain why we avoid string.ascii_letters

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
https://docs.python.org/3/library/string.html#string.ascii_letters

### CODE REVIEW

File: numpy/core/einsumfunc.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 __all__ = ['einsum', 'einsum_path']
 
-einsum_symbols = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ'
+einsum_symbols = string.ascii_letters
```
Comment (DimitriPapadopoulos) [inline numpy/core/einsumfunc.py:14]: I could discard `einsum_symbols` as well, and use `string.ascii_letters` everyhwere.

**Review by rossbar — COMMENTED** [review]
FWIW there have been related proposals elsewhere, see e.g. #23777. There, the main concern has been the import time of `string`.

**Comment by rkern** [comment]
Yes, I'd rather add a comment explaining that rather than introduce `string.ascii_letters`.

**Comment by DimitriPapadopoulos** [comment]
I see. OK, I'll write a comment instead.

**Comment by rossbar** [comment]
Thanks @DimitriPapadopoulos

**Pull request merged**